### PR TITLE
Move configurable fsgroup to GA

### PR DESCRIPTION
**Pull request by gnufied**
Move configurable fsgroup feature to GA. 

Also rename metric that reports time taken to change permissions - https://github.com/kubernetes/kubernetes/issues/104717

xref https://github.com/kubernetes/enhancements/issues/695

cc @msau42 @mattcary 

/sig storage

```release-note
Move ConfigurableFSGroupPolicy to GA
Rename metric volume_fsgroup_recursive_apply to volume_apply_access_control
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -816,7 +817,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	InTreePluginvSphereUnregister:                  {Default: false, PreRelease: featuregate.Alpha},
 	CSIMigrationOpenStack:                          {Default: true, PreRelease: featuregate.Beta},
 	InTreePluginOpenStackUnregister:                {Default: false, PreRelease: featuregate.Alpha},
-	ConfigurableFSGroupPolicy:                      {Default: true, PreRelease: featuregate.Beta},
+	ConfigurableFSGroupPolicy:                      {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.24
```
Comment (msau42) [inline pkg/features/kube_features.go:820]: Should be removed in 1.25 according to deprecation policy

**Comment by msau42** [comment]
Can you add a release note indicating that the metric name is changing? Do we have any e2e tests that the metric would need to update?

**Comment by dashpole** [comment]
/assign
/triage accepted
(lgtm from instrumentation)

File: pkg/features/kube_features.go (hunk 1 above)
Comment (gnufied) [reply pkg/features/kube_features.go]: fixed

**Comment by gnufied** [comment]
@msau42 updated release note. Also no, I don't think there are any e2e that depend on that metric.

**Comment by thockin** [comment]
I sitll see references to this gate in pkg/api/pod/util.go pkg/volume/volume_linux.go pkg/volume/volume_linux_test.go

**Comment by gnufied** [comment]
@thockin I thought until feature gate is really removed, some of the tests/code *can* carry featuregate references?  Not trying to use as an excuse but I saw similar examples elsewhere (i.e feature is being GAed but has some references in code).

**Comment by thockin** [comment]
Generally GA means to remove gated code

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**